hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -364,19 +363,24 @@ protected ClientConnection createSocketConnection(final Address address) throws
         }
     }
 
+    private int getBufferSize() {
+        int bufferSize = socketOptions.getBufferSize() * KILO_BYTE;
+        if (bufferSize <= 0) {
+            bufferSize = DEFAULT_BUFFER_SIZE_BYTE;
+        }
+        return bufferSize;
+    }
+
     @Override
-    public void destroyConnection(final Connection connection, final String reason, final Throwable cause) {
+    public void onClose(Connection connection) {
         Address endPoint = connection.getEndPoint();
-        ClientConnection conn = (ClientConnection) connection;
-        if (endPoint != null && connections.remove(endPoint, conn)) {
+
+        if (endPoint != null && connections.remove(endPoint, connection)) {
```

Comment:
Is there a possibility here where the listeners may not be notified in this case: a new connection to the same address is constructed and put into the map before this line is executed. Then the connection shall be removed but the listeners won't be notified, is this possible?